timesheet_automatic_remind/models/timesheet_employee_remind.py

```diff
@@ -0,0 +1,291 @@
+            self,
+            selected_date=False,
+            employee_id=False
+    ):
+        publicholiday = self.env['hr.holidays.public']
+        if not publicholiday.is_public_holiday(
+                selected_date,
+                employee_id
+        ) \
+                and not self.is_weekend(
+                    selected_date
+        ):
+            return True
+        else:
+            return False
+
+    @api.multi
+    def _create_remind_list_object(
+            self,
+            employee_remind_list=False
+    ):
+        for employee, datelist in employee_remind_list.items():
+            self.create(
+                {
+                    'employee_id': employee,
+                    'created_date': str(
+                        self._tz_offset_today(self.env.user.tz)
+                    ),
+                    'date_list': datelist
+                }
+            )
+
+    @api.multi
+    def _check_employee_workhours(
+            self,
+            employee_remind_list=False,
+            selected_date=False,
+            employee=False,
+            resethours=False,
+            allhours_expected=False,
+    ):
+        hranalytictmobj = self.env['hr.analytic.timesheet']
+        hrworklist = hranalytictmobj.search([
+            ('date',
+             '=',
+             str(selected_date)),
+            ('user_id',
+             '=',
+             employee.user_id.id)
+        ])
+        if hrworklist:
+            workhours_real = 0
+            for hrwork in hrworklist:
+                workhours_real += hrwork.unit_amount
+            allhours_real = workhours_real + resethours
+            if allhours_expected <= allhours_real:
+                pass
+            else:
+                employee_remind_list[employee.id].append(
+                    (
+                        0,
+                        False,
+                        {
+                            'reminder_date': str(selected_date)
+                        }
+                    )
+                )
+        else:
+            employee_remind_list[employee.id].append(
+                (
+                    0,
+                    False,
+                    {
+                        'reminder_date': str(selected_date)
+                    }
+                )
+            )
+
+    @api.model
+    def _start_remind(
+            self,
+            count_days=7,
+            workstart='9:00:00',
+            workend='18:00:00',
+            resethours=1
+    ):
+        employee_remind_list = {}
+        employee_list = self.env['hr.employee'].search([])
+        for employee in employee_list:
+            count = 0
+            selected_date = self._tz_offset_today(self.env.user.tz)
+            date_one_day = datetime.timedelta(days=1)
+            employee_remind_list[employee.id] = []
+            while count < count_days:
+                selected_date = selected_date - date_one_day
+                if self._check_weekend_or_holiday(
```

Comment:
Sometimes, a weekend day can be a working day. My suggestion is to inherit the `publicholiday` module to allow it to tell whether a day is a working day or not.